dbt/adapters/firebolt/impl.py

```diff
@@ -196,49 +198,25 @@ def raise_grant_error(self) -> None:
         Grant is not currently supported so this function
             raises an error.
         """
-        dbt.exceptions.raise_compiler_error(
+        raise CompilationError(
             'Firebolt does not support table-level permission grants.'
             ' Please remove grants section from the config.'
         )
 
     @available.parse_none
-    def sdk_column_list_to_firebolt_column_list(
-        self, columns: List[SDKColumn]
-    ) -> List[FireboltColumn]:
+    def resolve_special_columns(self, column: str) -> str:
         """
-        Extract and return list of FireboltColumns with names and data types
-        extracted from SDKColumns.
-        Args:
-          columns: list of Column types as defined in the Python SDK
+        Resolve special columns types that dbt is unable to parse natively.
         """
-        return [
-            FireboltColumn(
-                column=col.name, dtype=self.create_type_string(col.type_code)
-            )
-            for col in columns
-        ]
 
-    @available.parse_none
-    def create_type_string(self, type_code: Any) -> str:
-        """
-        Return properly formatted type string for SQL DDL.
-        Args: type_code is technically a type, but mypy complained that `type`
-        does not have an attribute `subtype`.
-        """
-        types = {
-            'str': 'TEXT',
-            'int': 'LONG',
-            'float': 'DOUBLE',
-            'date': 'DATE',
-            'datetime': 'DATE',
-            'bool': 'BOOLEAN',
-            'Decimal': 'DECIMAL',
-        }
-        type_code_str = '{}'
-        while isinstance(type_code, ARRAY):
-            type_code_str = f'ARRAY({type_code_str})'
-            type_code = type_code.subtype
-        return type_code_str.format(types[type_code.__name__])
+        if column.startswith('ARRAY'):
+            column = column[6:-1]
+            if column.endswith('NOT NULL'):
+                column = column[:-8]
+            if column.endswith('NULL'):
+                column = column[:-4]
+            return column.rstrip()
+        return column
```

Comment:
What if it's `INT NOT NULL`?

Reply:
Also what if it's `ARRAY(ARRAY(INT))`?

Reply:
`INT NOT NULL` is not possible since we read from the information schema, which contains column `is_nullable`. Both nullable and non-nullable ints show up as INT.
Fair point on the multi-dimension arrays. I wonder if we supported them at all in the previous implementation.

Reply:
OK, makes sense, please verify nested arrays support then

Reply:
I've reworked array processing and added unit tests to verify the parsing is correct.